plugins/netstat/helper/netstathelper.cpp

```diff
@@ -17,30 +25,121 @@ KAuth::ActionReply NetstatHelper::query(const QVariantMap)
 
     QProcess    netstat;
     QStringList netstatArgs("-ntuap");
-    qDebug() << "run" << NETSTAT_BINARY_PATH << netstatArgs;
+    QString executable = mHasSS ? QStringLiteral("ss")
+                      : mHasNetstat ? QStringLiteral("netstat")
+                      : QString();
+
+    if (executable.isEmpty()) {
+        qWarning() << "No iproute or net-tools installed, can't run.";
+        KAuth::ActionReply::HelperErrorReply(-2);
+        return {};
+    }
+
+    qDebug() << "run" << executable << netstatArgs;
+
+    netstat.start(executable, netstatArgs, QIODevice::ReadOnly);
 
-    netstat.start(NETSTAT_BINARY_PATH, netstatArgs, QIODevice::ReadOnly);
     if (netstat.waitForStarted())
         netstat.waitForFinished();
 
     int exitCode(netstat.exitCode());
 
     if(0 != exitCode)
     {
-        qWarning() << "netstat command exit with code: " << exitCode;
+        qWarning() << executable << "command exit with code: " << exitCode;
 
         reply=KAuth::ActionReply::HelperErrorReply(exitCode);
         reply.addData("response", netstat.readAllStandardError());
     } else {
         QVariantList connections = parseOutput(netstat.readAllStandardOutput());
-//        qDebug() << connections;
         reply.addData("connections", connections);
     }
 
     return reply;
 }
 
 QVariantList NetstatHelper::parseOutput(const QByteArray &netstatOutput)
+{
+    if (mHasSS) {
+        return parseSSOutput(netstatOutput);
+    } else if (mHasNetstat) {
+        return parseNetstatOutput(netstatOutput);
+    }
+    return {};
+}
+
+QVariantList NetstatHelper::parseSSOutput(const QByteArray &netstatOutput)
+{
+   QString rawOutput = netstatOutput;
+    QStringList outputLines = rawOutput.split("\n");
+
+    QVariantList connections;
+
+    // discard lines.
+    while (outputLines.size()) {
+        if (outputLines.first().indexOf("Recv-Q")) {
+            outputLines.removeFirst();
+            break;
+        }
+        outputLines.removeFirst();
+    }
+
+    // can't easily parse because of the spaces in Local and Peer AddressPort.
+    QStringList headerLines = {
+        QStringLiteral("Netid"),
+        QStringLiteral("State"),
+        QStringLiteral("Recv-Q"),
+        QStringLiteral("Send-Q"),
+        QStringLiteral("Local Address:Port"),
+        QStringLiteral("Peer Address:Port"),
+        QStringLiteral("Process"),
+    };
+
+    // Extract Information
+    for (auto line : outputLines)
+    {
+        QStringList values = line.split(" ", Qt::SkipEmptyParts);
```

Comment:
This is causing the Travis build to fail. Qt 5.9 is being used there.
I wasn't able to find in the doc when this feature was added. Maybe we should consider a workaround.

Reply:
Is 5.9 really a requirement for any user?